roles/openshift_on_openstack/tasks/main.yml

```diff
@@ -188,50 +201,73 @@
     dest: "{{ osev3_yml_path }}"
     backup: yes
 
-
-# Get some debug information that makes it easier to file bugs.
-- name: Running commands that create debug information needed to open issues
-  shell: "{{ item }}"
-  register: debug_information
-  with_items:
-    - "printf \"$(uname -srm)\n$(cat /etc/os-release)\n\""
-    - "ansible --version"
-    - "cd openshift-ansible ; git describe"
-    - "rpm -q atomic-openshift-utils openshift-ansible"
-  changed_when: false
+- name: Copy {{ osev3_yml_path }} to the artifacts directory
+  synchronize:
+    src: "{{ osev3_yml_path }}"
+    dest: "{{ artifacts_directory }}"
+    mode: pull
   ignore_errors: true
 
-# Print the debug information out.
-- name: Printing the debug information to the screen
-  debug:
-    msg: "{{ item['stdout_lines'] }}"
-  with_items: "{{ debug_information['results'] }}"
-
 # Set the path for the prerequisite log file.
 - name: Creating a variable for the prerequisite log file
   set_fact:
     prerequisite_log: "{{ ansible_user_dir }}/openshift_prerequisites.log"
 
-# Run the Ansible playbook that checks the prerequisites in OpenStack.
-- name: Checking the prerequisites for cluster provisioning in OpenStack
-  # Using two inventory flags here as the README instructs.
-  shell: "{{ ansible_playbook }} -vvv --user openshift -i inventory -i {{ inventory_py }} {{ openshift_openstack_dir }}/openshift-cluster/prerequisites.yml 2>&1 >> {{ prerequisite_log }}"
-  args:
-    # Use bash to get the posix style redirects.
-    executable: /bin/bash
+- name: OpenShift prerequisite block
+  block:
+    # Run the Ansible playbook that checks the prerequisites in OpenStack.
+    - name: Checking the prerequisites for cluster provisioning in OpenStack
+      # Using two inventory flags here as the README instructs.
+      shell: "{{ ansible_playbook }} -vvv --user openshift -i inventory -i {{ inventory_py }} {{ openshift_openstack_dir }}/openshift-cluster/prerequisites.yml 2>&1 >> {{ prerequisite_log }}"
+      args:
+        # Use bash to get the posix style redirects.
+        executable: /bin/bash
+  always:
+    - name: Copy the log file to the artifacts directory
+      synchronize:
+        src: "{{ prerequisite_log }}"
+        dest: "{{ artifacts_directory }}"
+        mode: pull
+      ignore_errors: true
+
+    # Get some debug information that makes it easier to file OpenShift bugs.
+    - name: Running commands that create debug information needed to open issues
+      shell: |
+        printf $(uname -srm)\n$(cat /etc/os-release)\n > debug_information.txt
```

Comment:
You might want to explicitly path the debug output such as `/home/cloud-user/ocp-scale/logs/debug_information.txt`  Also then you would just consistently synchronize the entire directory ex `/home/cloud-user/ocp-scale/logs/`  in fact i would co-locate all debug and log info into one directory so it would be easy to find and not clutter/pollute the user home directory.

This would be easy to find all your active debug info incase of job failure or just general logs surround the job install in one convenient directory.

Reply:
Ok sure, lets get this wrapped up quickly then so we can open another pr and get that fixed too.